[Bug]: internal/msgstream unittest panics randomly

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:535441d
- Deployment mode(standalone or cluster): both
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

`msgstream` package unit test panics randomly

### Expected Behavior

Unit tests passes

### Steps To Reproduce

```markdown
Run unit test in msgstream package multiple times
```


### Anything else?
```
--- FAIL: TestStream_PulsarMsgStream_InsertRepackFunc (0.17s)
panic: topic persistent://public/default/dGOcPubs, subscription FZhxFNgl: server error: MetadataError: Consumer not found [recovered]
	panic: topic persistent://public/default/dGOcPubs, subscription FZhxFNgl: server error: MetadataError: Consumer not found

goroutine 546 [running]:
testing.tRunner.func1.2(0x16cfec0, 0xc000895db0)
	/usr/local/go/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc00075db00)
	/usr/local/go/src/testing/testing.go:1146 +0x4b6
panic(0x16cfec0, 0xc000895db0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/milvus-io/milvus/internal/util/mqclient.(*PulsarConsumer).Close.func1()
	/home/silverxia/workspace/milvus/internal/util/mqclient/pulsar_consumer.go:113 +0xc5
sync.(*Once).doSlow(0xc0008a39c8, 0xc0000d9b40)
	/usr/local/go/src/sync/once.go:68 +0xec
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
github.com/milvus-io/milvus/internal/util/mqclient.(*PulsarConsumer).Close(0xc0008a3980)
	/home/silverxia/workspace/milvus/internal/util/mqclient/pulsar_consumer.go:108 +0x65
github.com/milvus-io/milvus/internal/msgstream.(*mqMsgStream).Close(0xc000281d40)
	/home/silverxia/workspace/milvus/internal/msgstream/mq_msgstream.go:226 +0x173
github.com/milvus-io/milvus/internal/msgstream.TestStream_PulsarMsgStream_InsertRepackFunc(0xc00075db00)
	/home/silverxia/workspace/milvus/internal/msgstream/mq_msgstream_test.go:557 +0x858
testing.tRunner(0xc00075db00, 0x18e5ee8)
	/usr/local/go/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x2b3

```

--- a/internal/datanode/data_node_test.go
+++ b/internal/datanode/data_node_test.go
@@ -143,8 +143,6 @@ func TestDataNode(t *testing.T) {
 		err = node1.Start()
 		assert.Nil(t, err)
 		defer func() {
-			// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-			time.Sleep(200 * time.Millisecond)
 			err := node1.Stop()
 			assert.Nil(t, err)
 		}()
@@ -338,8 +336,6 @@ func TestDataNode(t *testing.T) {
 			if i <= 2 {
 				err = node.flowgraphManager.addAndStart(node, &datapb.VchannelInfo{CollectionID: 1, ChannelName: test.dmChannelName})
 				assert.Nil(t, err)
-				// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-				time.Sleep(200 * time.Millisecond)
 				vchanNameCh <- test.dmChannelName
 			}
 		}
@@ -413,8 +409,6 @@ func TestWatchChannel(t *testing.T) {
 		exist := node.flowgraphManager.exist(ch)
 		assert.True(t, exist)
 
-		// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-		time.Sleep(200 * time.Millisecond)
 		err = kv.RemoveWithPrefix(fmt.Sprintf("%s/%d", Params.DataNodeCfg.ChannelWatchSubPath, node.NodeID))
 		assert.Nil(t, err)
 		//TODO there is not way to sync Release done, use sleep for now

--- a/internal/datanode/flow_graph_manager_test.go
+++ b/internal/datanode/flow_graph_manager_test.go
@@ -19,7 +19,6 @@ package datanode
 import (
 	"context"
 	"testing"
-	"time"
 
 	"github.com/milvus-io/milvus/internal/proto/datapb"
 	"github.com/milvus-io/milvus/internal/proto/internalpb"
@@ -46,8 +45,6 @@ func TestFlowGraphManager(t *testing.T) {
 
 	fm := newFlowgraphManager()
 	defer func() {
-		// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-		time.Sleep(200 * time.Millisecond)
 		fm.dropAll()
 	}()
 	t.Run("Test addAndStart", func(t *testing.T) {
@@ -62,8 +59,6 @@ func TestFlowGraphManager(t *testing.T) {
 		assert.NoError(t, err)
 		assert.True(t, fm.exist(vchanName))
 
-		// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-		time.Sleep(200 * time.Millisecond)
 		fm.dropAll()
 	})
 
@@ -79,8 +74,6 @@ func TestFlowGraphManager(t *testing.T) {
 		assert.NoError(t, err)
 		assert.True(t, fm.exist(vchanName))
 
-		// TODO: wait for reconnecting to Pulsar, delete sleep after Seek wouldn't lead to disconnect with Pulsar
-		time.Sleep(200 * time.Millisecond)
 		fm.release(vchanName)
 
 		assert.False(t, fm.exist(vchanName))

--- a/internal/util/mqclient/pulsar_consumer.go
+++ b/internal/util/mqclient/pulsar_consumer.go
@@ -17,11 +17,15 @@
 package mqclient
 
 import (
+	"context"
 	"sync"
+	"time"
 	"unsafe"
 
 	"github.com/apache/pulsar-client-go/pulsar"
 	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/util/retry"
+	"go.uber.org/zap"
 )
 
 // PulsarConsumer consumes from pulsar
@@ -108,8 +112,12 @@ func (pc *PulsarConsumer) Close() {
 	pc.closeOnce.Do(func() {
 		defer pc.c.Close()
 		// Unsubscribe for the consumer
-		err := pc.c.Unsubscribe()
+		err := retry.Do(context.Background(), func() error {
+			//TODO need to check error retryable
+			return pc.c.Unsubscribe()
+		}, retry.MaxSleepTime(50*time.Millisecond), retry.Attempts(3))
 		if err != nil {
+			log.Error("failed to unsubscribe", zap.String("subscription", pc.Subscription()), zap.Error(err))
 			panic(err)
 		}
 		close(pc.closeCh)

